src/pages/Admin/Charity/index.tsx

```diff
@@ -51,3 +54,30 @@ export default function Charity() {
     </Routes>
   );
 }
+
+function createLinkGroups(endowId: number): LinkGroup[] {
+  return [
+    {
+      links: [
+        LINKS[adminRoutes.index],
+        LINKS[adminRoutes.withdraws],
+        LINKS[adminRoutes.contributions],
+      ],
+    },
+    {
+      title: "Invest",
+      links: [LINKS[adminRoutes.invest], LINKS[adminRoutes.settings]],
+    },
+    { title: "Profile", links: [LINKS[adminRoutes.edit_profile]] },
+    {
+      title: "Manage",
+      links: [
+        LINKS[adminRoutes.proposals],
+        {
+          ...LINKS[adminRoutes.widget_config],
+          to: `${LINKS[adminRoutes.widget_config].to}/${endowId}`,
```

Comment:
`/{endowId}` *is* necessary for the widget configurer to know what is the default endowment to create the widget for, see PR https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1746 for more details.

Reply:
if `<WidgetConfigurer/>` needs to know about default `endowmentId`, could just get it by itself 
```ts
//WidgetConfigurer
const { id } = useAdminResources()
```

Reply:
Yes, but then there would be no way to tell the widget that the endowment to donate to will be selected using its (widget's) combobox:
- if `/{endowId}` is present -> just display the endowment name
- if no ID param is present -> display the endowment combobox

Reply:
Anyway, that's how the widget configurer currently works. Any changes to its function are out of scope for this PR.

Reply:
@misicnenad, you are using this `<WidgetConfigurer/>` in the context of `/admin/:id`
just look at how you use your `createLinkGroups` - literally just passing `id` of current admin page 

```ts
 const { id } = useAdminResources();
 const linkGroups = useMemo(() => createLinkGroups(id), [id]); 
  
```

> Yes, but then there would be no way to tell the widget that the endowment to donate to will be selected using its (widget's) combobox:

there's even no need to for `configurer/{endowId}` 
If I am currently in `admin/1` is there a way for me to navigate to `admin/{1}/widget-configurer/{2}` without explicitly typing it in a browser?

> if no ID param is present -> display the endowment combobox

there's always `ID` param, in `admin/:id`, and `<WidgetConfigurer/>` should display the endowment's name by default. If user intends to configure the widget to donate to other endowment - could just select other in dropdown.


> Anyway, that's how the widget configurer currently works. Any changes to its function are out of scope for this PR.

imo, we should adjust a component's `api` to suite the page/use case it is used into - not adjust page implementation just to suit already merge component `api`

Reply:
> If I am currently in admin/1 is there a way for me to navigate to admin/{1}/widget-configurer/{2} without explicitly typing it in a browser?

That's a valid point @SovereignAndrey 

> there's always ID param, in admin/:id, and <WidgetConfigurer/> should display the endowment's name by default. If user intends to configure the widget to donate to other endowment - could just select other in dropdown

This goes against the original requirements https://app.clickup.com/t/865bp9483?comment=90040003682783&threadedComment=90080010341495

> there's always ID param, in admin/:id, and <WidgetConfigurer/> should display the endowment's name by default. If user intends to configure the widget to donate to other endowment - could just select other in dropdown
> ...
> imo, we should adjust a component's api to suite the page/use case it is used into - not adjust page implementation just to suit already merge component api

If we're talking about an internal implementation, then I agree with this statement.
But when we're talking about something external. like UI/UX (whether an endowment selector or plain text of their name appears), I'd be more careful with these changes and generally leave these types of decisions to the product owners - in these cases, the changes are best left for a separate PR.

Reply:
@SovereignAndrey @ap-justin **this PR is just about updating the existing admin pages _as they are_ to use the new Sidebar. Whether the specific pages' logic should be updated, although a valid concern, should be treated as a completely separate issue.**

Reply:
@misicnenad 

> This goes against the original requirements ([link to comment](https://app.clickup.com/t/865bp9483?comment=90040003682783&threadedComment=90080010341495)) and is therefore out of scope of this PR.

I believe this is for a widget being used as a standalone page `/widget-configurer/:id`

but in the context of an `admin/endowment` - an admin member should only concern itself on configuring their endowment's widget as you mentioned `A charity shouldn't need to have to navigate to Widget configurer with a charity that's not them set as default`

 that is 

```tsx
//widget-configurer used in the context of an admin
const { id } = useAdminResources()
<WidgetConfigurer defaultid={id} /> 
```

if widget configurer is to be used as a standalone page - like `user/create-widget` then param` endowId` makes sense and could be implemented

```tsx
//create widget page
const {endowIdFromParam} = useParams()
<WidgetConfigurer defaultid={endowIdFromParam} /> 
```

I am questioning the validity of `endowId` param for the use in the admin pages

so to avoid creating a `createLinkGroups` helper that just reassigns/reshuffles `constant LINKS` values 
```
        LINKS[adminRoutes.proposals],
        {
          ...LINKS[adminRoutes.widget_config],
          to: `${LINKS[adminRoutes.widget_config].to}/${endowId}`,
```
and recommends to just use `LINKS` directly to unneccesary layers of abstraction

Reply:
@ap-justin this is just how the current implementation of widget configurer is used. Updating the widget configurer **is out of scope of this PR. This PR only concerns itself with updating the existing pages as they are to use the new Sidebar**. Lets create a new ticket to update the widget configurer and move this discussion there.

Reply:
@misicnenad, since widget-configurer forces to make unnecessary layers of reassignment in the implementation of this layout

```ts
//createLinkGroup
  return [
    {
      links: [
        LINKS[adminRoutes.index],
        LINKS[adminRoutes.withdraws],
        LINKS[adminRoutes.contributions],
      ],
    },
    {
      title: "Invest",
      links: [LINKS[adminRoutes.invest], LINKS[adminRoutes.settings]],
    },
    { title: "Profile", links: [LINKS[adminRoutes.edit_profile]] },
    {
      title: "Manage",
      links: [
        LINKS[adminRoutes.proposals],
        {
          ...LINKS[adminRoutes.widget_config],
          to: `${LINKS[adminRoutes.widget_config].to}/${endowId}`,
```
I suggest we don't include it in this PR and layout components be implemented to just use plain `const LINKS`

Reply:
And I suggest we do that as part of a separate PR. @ap-justin please, this is not relevant to the task at hand and the implementation is not that complex either to execute or to understand. This is not that big of a deal that it has to hold back the whole PR.

Reply:
> If you want to discuss how to optimize widget configurer

I don't know what you mean by `optimize widget-configurer` I just want these code not merged in this `<Layout/>` implementation`
```
//createLinkGroup
  return [
    {
      links: [
        LINKS[adminRoutes.index],
        LINKS[adminRoutes.withdraws],
        LINKS[adminRoutes.contributions],
      ],
    },
    {
      title: "Invest",
      links: [LINKS[adminRoutes.invest], LINKS[adminRoutes.settings]],
    },
    { title: "Profile", links: [LINKS[adminRoutes.edit_profile]] },
    {
      title: "Manage",
      links: [
        LINKS[adminRoutes.proposals],
        {
          ...LINKS[adminRoutes.widget_config],
          to: `${LINKS[adminRoutes.widget_config].to}/${endowId}`,
```

> optimize widget configurer, that'd be great, but lets do that as part of a separate and relevant task/PR, not inside a task to add sidebar to existing pages.

that's why I'am asking not to include `<WidgetConfigurer/>` since its implementation is just for standalone page [see comment](https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1804#discussion_r1112635054)

Reply:
@misicnenad - see simple solution #1846

Reply:
> that's why I'am asking not to include `<WidgetConfigurer/>`

@ap-justin this is not my decision to make, you should talk to @SovereignAndrey

Reply:
Will remove link to widget configurer for now, as it was commented out on `RC` branch anyway.